src/renderer/components/+extensions/extensions.tsx

```diff
@@ -308,22 +303,77 @@ async function installOnDrop(files: File[]) {
   await requestInstalls(files.map(({ path }) => path));
 }
 
+async function requestInstallVersion(url: URLParse): Promise<string | null> {
+  const disposer = ExtensionInstallationStateStore.startPreInstall();
+  let json: Record<string, any>;
+
+  try {
+    const ticket = downloadFile({ url, gzip: false, timeout: 60_000 });
+    const res = await ticket.promise;
+
+    json = JSON.parse(res.toString());
+  } catch (error) {
+    console.error(error);
+    Notifications.error("Failed to get registry information for that extension");
+    disposer();
+
+    return null;
+  }
+
+  let selected: string;
+  const options = Object.keys(json.versions)
+    .map(version => new SemVer(version, { loose: true, includePrerelease: true }))
+    .sort((a, b) => b.compare(a))
+    .map(a => a.format());
+
+  const removeNotification = Notifications.info(
+    <div className="InstallingExtensionNotification flex gaps align-center">
+      <div className="flex column gaps">
+        <p>Install extension <b>{json.name}</b>?</p>
+        <p>Description: <em>{json.description}</em></p>
+        <Select
+          menuClass="NotificationSelect"
+          options={options}
+          placeholder="Select version to install"
+          onChange={(selection: SelectOption<string>) => selected = selection.value}
+        />
+      </div>
+      <Button autoFocus label="Install" onClick={async () => {
+        removeNotification();
+
+        const url = json.versions[selected].dist.tarball;
+        const fileName = path.basename(url);
+        const { promise: dataP } = downloadFile({ url, timeout: 60000 /*1m*/ });
+
+        await requestInstall({ fileName, dataP }, disposer);
+      }} />
+    </div>
+  );
+}
+
 async function installFromUrlOrPath(installPath: string) {
   const fileName = path.basename(installPath);
 
   try {
     // install via url
     // fixme: improve error messages for non-tar-file URLs
     if (InputValidators.isUrl.validate(installPath)) {
-      const disposer = ExtensionInstallationStateStore.startPreInstall();
-      const { promise: filePromise } = downloadFile({ url: installPath, timeout: 60000 /*1m*/ });
-      const data = await filePromise;
+      const url = new URLParse(installPath);
+
+      if (url.hostname === "registry.npmjs.com") {
```

Comment:
It would be nice to be able to just specify `@company/ext-name` and have Lens try to get it from `registry.npmjs.com` by default.